apps/studio/src/server/modules/user/user.router.ts

```diff
@@ -0,0 +1,296 @@
+          "ActiveUser.id",
+          "ActiveUser.email",
+          "ActiveUser.name",
+          "ActiveUser.lastLoginAt",
+          eb.ref("ActiveResourcePermission.role").as("role"),
+        ])
+        .executeTakeFirst()
+
+      if (!result) {
+        throw new TRPCError({
+          code: "NOT_FOUND",
+          message: "User not found",
+        })
+      }
+
+      return result
+    }),
+
+  list: protectedProcedure
+    .input(listUsersInputSchema)
+    .output(listUsersOutputSchema)
+    .query(async ({ ctx, input: { siteId, adminType, offset, limit } }) => {
+      await validatePermissionsForManagingUsers({
+        siteId,
+        userId: ctx.user.id,
+        action: "read",
+      })
+
+      return getUsersQuery({ siteId, adminType })
+        .orderBy("ActiveUser.lastLoginAt", sql.raw(`DESC NULLS LAST`))
+        .select((eb) => [
+          "ActiveUser.id",
+          "ActiveUser.email",
+          "ActiveUser.name",
+          "ActiveUser.lastLoginAt",
+          eb.ref("ActiveResourcePermission.role").as("role"),
+        ])
+        .limit(limit)
+        .offset(offset)
+        .execute()
+    }),
+
+  count: protectedProcedure
+    .input(countUsersInputSchema)
+    .output(countUsersOutputSchema)
+    .query(async ({ ctx, input: { siteId, adminType } }) => {
+      await validatePermissionsForManagingUsers({
+        siteId,
+        userId: ctx.user.id,
+        action: "read",
+      })
+
+      const result = await getUsersQuery({ siteId, adminType })
+        .select((eb) => [eb.fn.countAll().as("count")])
+        .executeTakeFirstOrThrow()
+
+      return Number(result.count)
+    }),
+
+  hasInactiveUsers: protectedProcedure
+    .input(hasInactiveUsersInputSchema)
+    .output(hasInactiveUsersOutputSchema)
+    .query(async ({ ctx, input: { siteId } }) => {
+      await validatePermissionsForManagingUsers({
+        siteId,
+        userId: ctx.user.id,
+        action: "read",
+      })
+
+      const ninetyDaysAgo = new Date()
+      ninetyDaysAgo.setDate(ninetyDaysAgo.getDate() - 90)
+
+      const result = await getUsersQuery({ siteId, adminType: "agency" })
+        .select((eb) => [eb.fn.countAll().as("count")])
+        .where("ActiveUser.lastLoginAt", "is not", null)
+        .where("ActiveUser.lastLoginAt", "<", ninetyDaysAgo)
+        .executeTakeFirstOrThrow()
+
+      return Number(result.count) > 0
+    }),
+
+  update: protectedProcedure
+    .input(updateUserInputSchema)
+    .output(updateUserOutputSchema)
+    .mutation(async ({ ctx, input: { siteId, userId, role } }) => {
+      await validatePermissionsForManagingUsers({
+        siteId,
+        userId: ctx.user.id,
+        action: "update",
+      })
+
+      if (userId === ctx.user.id) {
+        throw new TRPCError({
+          code: "FORBIDDEN",
+          message: "You cannot update your own role",
+        })
+      }
+
+      const user = await db
+        .selectFrom("User")
+        .where("id", "=", userId)
```

Comment:
[issue] should we also be filtering so that hte users are not soft deleted?

Reply:
good point, updated in https://github.com/opengovsg/isomer/pull/1118/commits/28d4027a161319f9341700a0741245b495b6bc23